test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+	for i, c := range pod.Status.ContainerStatuses {
+		csMap[c.Name] = &pod.Status.ContainerStatuses[i]
+	}
+	for _, ci := range tcInfo {
+		gomega.Expect(csMap).Should(gomega.HaveKey(ci.Name))
+		cs := csMap[ci.Name]
+		tc, _ := makeTestContainer(ci)
+		gomega.Expect(tc.Resources).To(gomega.Equal(*cs.Resources))
+		// gomega.Expect(cs.RestartCount).To(gomega.Equal(ci.RestartCount))
+	}
+}
+
+func isPodOnCgroupv2Node(ctx context.Context, f *framework.Framework, pod *v1.Pod, flagError bool) bool {
+	// Determine if pod is running on cgroupv2 or cgroupv1 node
+	// TODO(vinaykul,InPlacePodVerticalScaling): Is there a better way to determine this?
+	ginkgo.By("Sleep " + PollInterval.String() + " seconds")
+	time.Sleep(PollInterval)
+	cgroupv2File := "/sys/fs/cgroup/cgroup.controllers"
+	mycmd := fmt.Sprintf("ls %s", cgroupv2File)
+	stdout, stderr, err := e2epod.ExecShellInPodWithFullOutput(ctx, f, pod.Name, mycmd)
+	if flagError {
+		framework.ExpectNoError(err, "failed to determine if pod is running on cgroupv2 or cgroupv1 node, connection to container failed")
+	}
+	if stderr == "ls: cannot access '/sys/fs/cgroup/cgroup.controllers': No such file or directory" {
+		framework.Logf("Pod is running on cgroupv1")
+		return false
+	}
+	if stderr == "" {
+		if stdout != "" {
+			framework.Logf("Pod is running on cgroupv2: " + stdout)
+			return true
+		}
+	}
+	framework.Logf("Assuming Pod is running on cgroupv2: " + stdout + "," + stderr)
+	return true
+}
+
+func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework, pod *v1.Pod, tcInfo []TestContainerInfo, flagError bool) bool {
+	podOnCgroupv2Node := isPodOnCgroupv2Node(ctx, f, pod, true)
+	cgroupMemLimit := Cgroupv2MemLimit
+	cgroupCPULimit := Cgroupv2CPULimit
+	cgroupCPURequest := Cgroupv2CPURequest
+	if !podOnCgroupv2Node {
+		cgroupMemLimit = CgroupMemLimit
+		cgroupCPULimit = CgroupCPUQuota
+		cgroupCPURequest = CgroupCPUShares
+	}
+	verifyCgroupValue := func(cName, cgPath, expectedCgValue string) bool {
+		ginkgo.By("Sleep " + PollInterval.String() + " seconds")
+		time.Sleep(PollInterval)
+		mycmd := fmt.Sprintf("head -n 1 %s", cgPath)
+		cgValue, _, err := ExecCommandInContainerWithFullOutput(ctx, f, pod.Name, cName, "/bin/sh", "-c", mycmd)
+		framework.Logf("Namespace %s Pod %s Container %s - looking for cgroup value %s in path %s",
+			pod.Namespace, pod.Name, cName, expectedCgValue, cgPath)
+		if flagError {
+			framework.ExpectNoError(err, "failed to find expected value '%s' in container cgroup '%s'",
+				expectedCgValue, cgPath)
+		}
+		cgValue = strings.Trim(cgValue, "\n")
+		if flagError {
+			gomega.Expect(cgValue).Should(gomega.Equal(expectedCgValue), "cgroup value")
+		}
+		if cgValue != expectedCgValue {
+			return false
+		}
+		return true
+	}
+	for _, ci := range tcInfo {
+		if ci.Resources == nil {
+			continue
+		}
+		tc, _ := makeTestContainer(ci)
+		if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
+			var cpuShares int64
```

Comment:
nit
```suggestion
			var expectedCPUShares int64
```

Reply:
Done, in addition did same with expectedCPULimit and expectedMemLimit